crates/proof-of-sql/src/sql/ast/group_by_util.rs

```diff
@@ -107,19 +107,16 @@ pub(super) fn sum_aggregate_column_by_index_counts<'a, S: Scalar>(
     indexes: &[usize],
 ) -> &'a [S] {
     match column {
-        Column::Boolean(col) => sum_aggregate_slice_by_index_counts(alloc, col, counts, indexes),
         Column::SmallInt(col) => sum_aggregate_slice_by_index_counts(alloc, col, counts, indexes),
         Column::Int(col) => sum_aggregate_slice_by_index_counts(alloc, col, counts, indexes),
         Column::BigInt(col) => sum_aggregate_slice_by_index_counts(alloc, col, counts, indexes),
         Column::Int128(col) => sum_aggregate_slice_by_index_counts(alloc, col, counts, indexes),
-        Column::Decimal75(_, _, _col) => {
-            todo!("aggregation over decimals not yet supported")
+        Column::Decimal75(_, _, col) => {
+            sum_aggregate_slice_by_index_counts(alloc, col, counts, indexes)
         }
         Column::Scalar(col) => sum_aggregate_slice_by_index_counts(alloc, col, counts, indexes),
-        Column::VarChar(_) => panic!("Cannot sum varchar columns"),
-        Column::TimestampTZ(_, _, _) => {
-            panic!("Cannot sum varchar columns")
-        }
+        // The following should never be reached because the `SUM` function can only be applied to numeric types.
+        _ => unreachable!("SUM can not be applied to non-numeric types"),
```

Comment:
Could you actually lay out the cases for this? Just so when we add new types it'll cause a compiler error until we update this code?

```suggestion
        Column::VarChar(_) | Column::TimestampTZ(_) | Column::Boolean(_) => unreachable!("SUM can not be applied to non-numeric types"),
```

Reply:
Sure!